src/main/java/com/google/api/gax/bundling/ThresholdBundler.java

```diff
@@ -49,60 +49,85 @@
  */
 public final class ThresholdBundler<E> {
 
-  private ImmutableList<BundlingThreshold<E>> thresholdPrototypes;
+  private class ReleaseResourcesFunction implements Function<Object, Void> {
```

Comment:
Shouldn't this be Function<ApiFuture<?>, Void>?

If this isn't possible, maybe change the ThresholdBundleReceiver.processBundle to return ApiFuture<Void> and use that every where? I tried reading the spec on how <?> behaves and got completely overwhelmed....